realm/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -143,7 +143,7 @@ public int hashCode() {
         result = 31 * result + realmFileName.hashCode();
         result = 31 * result + canonicalPath.hashCode();
         result = 31 * result + (key != null ? Arrays.hashCode(key) : 0);
-        result = 31 * result + schemaVersion;
+        result = 31 * result + (int)schemaVersion;
```

Comment:
From an academic point, this is a bit dangerous, but in practise no one should get into problem here.